src/QueryDataTable.php

```diff
@@ -631,7 +631,9 @@ protected function defaultOrdering(): void
             ->each(function ($orderable) {
                 $column = $this->resolveRelationColumn($orderable['name']);
 
-                if ($this->hasOrderColumn($column)) {
+                if ($this->hasOrderColumn($orderable['name'])) {
```

Comment:
Having some doubts about what will happen if the relationship is not joined in the query. Maybe we should switch the check so it will be a fallback when the eager loading check failed?

```PHP
                if ($this->hasOrderColumn($column)) {
                    $this->applyOrderColumn($column, $orderable);
                } elseif ($this->hasOrderColumn($orderable['name'])) {
                    $this->applyOrderColumn($orderable['name'], $orderable);
```

Reply:
if the relation is not eager load, for an EloquentDatatable, it would return the column name directly : 

EloquentDataTable:157

```
if ($this->isNotEagerLoaded($relation)) {
        return $column;
}
```

so  as `$column == $orderable['name']` there is no difference between the if

And the other hand even if the relation **IS**  join the query,I think that a custom order should always be applied (if not, why specified a custom orderColumn()) because if you don't want to join a table for any reason for your order, you would not be able to do that	

exemple : 

```php
protected function setUp(): void
    {
        parent::setUp();

        $this->app['router']->get('/relations/belongsTo', function (DataTables $datatables) {
            return $datatables->eloquent(Post::select('posts.*')
											// without eagerload
                                             //->with('user')
                                             )
                              ->orderColumn('user.id', function ($query, $order) {
								// Order with a subquery instead of a join
                                  return $query->orderBy(DB::raw('(select id from users where users.id = posts.user_id)'),  $order == 'desc' ? 'asc' : 'desc');
                              })
                              ->editColumn('user.id', function ($post) {
                                  return $post->user->id;
                              })
                              ->toJson();
        });
    }
```